src/hooks/useDimensions.ts

```diff
@@ -0,0 +1,29 @@
+import { MutableRefObject, useLayoutEffect, useMemo, useState } from "react";
```

Comment:
I wanna advocate for following this pattern for hooks so we can colocate the tests and hooks in the same folder. 
```bash
hooks/useDimensions/index.ts
hooks/useDimensions/useDimensions.test.ts
```